File: Final Project/Code/DD_SVMReducedModel_05_02_2023.py
# -*- coding: utf-8 -*-
"""
Created on Sun May  7 16:23:18 2023

@author: ddinh
"""

#%%===== Load Appropriate Libraries ===============================================================================
import pandas as pd  
#import numpy as np 
#import os
#import matplotlib.pyplot as plt  
from sklearn.model_selection import train_test_split  
from sklearn.svm import SVC  
from sklearn.metrics import classification_report, confusion_matrix  
#from sklearn.model_selection import StratifiedShuffleSplit
#from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import cross_val_score, cross_val_predict
#from sklearn.model_selection import KFold
#%%===== Loading data pre-processing and re arranging   ===========================================================
df1 = pd.read_csv("C:/Users/ddinh/OneDrive/Documents/EXST 7087/Final Project/Data/DD_ReducedCleanCompetitiveCareer_05_01_2023.csv")

X = df1.drop('Results', axis=1)  
y = df1['Results'] 

#%%===============Using Linear Kernel ===============================================================
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.20)  
svclassifier = SVC(kernel='linear')  
svclassifier.fit(X_train, y_train)

y_pred = svclassifier.predict(X_test)  
print(y_pred)
print(confusion_matrix(y_test,y_pred))  
print(classification_report(y_test,y_pred))  



#%%========================= Cross Validating =====================================================
cross_val_scores = cross_val_score(svclassifier, X_train, y_train, cv=5)


# View the cross-validated scores
print("Cross-Validated Accuracy Scores:", cross_val_scores)
print("Mean Accuracy:", cross_val_scores.mean())

# Get cross-validated predictions
cross_val_predictions = cross_val_predict(svclassifier, X_train, y_train, cv=5)

# Calculate the confusion matrix
conf_mat = confusion_matrix(y_train, cross_val_predictions)

# Print the confusion matrix
print("Confusion Matrix:", conf_mat)

File: Final Project/Code/DD_SVMReducedModel_05_02_2023.R
library(e1071)
library(caret)
library(kernlab)


df= read.csv("C:/Users/ddinh/OneDrive/Documents/EXST 7087/Final Project/Data/DD_ReducedCleanCompetitiveCareer_05_01_2023.csv")

# Split the data into training and testing sets
set.seed(123)  # Set a seed for reproducibility
train_indices <- sample(nrow(df), nrow(df) * 0.8)  # 70% for training
train_data <- df[train_indices, ]
test_data <- df[-train_indices, ]

# Train the SVM model with all columns except the first as predictors
svm_model <- svm(Results ~ ., data = train_data, type = "C-classification", kernel = "linear")

# Make predictions on the test data
svm_pred <- predict(svm_model, newdata = test_data)

# Evaluate the SVM model
accuracy <- sum(svm_pred == test_data$Results) / length(svm_pred)
print(paste("Accuracy:", accuracy))


# Perform cross-validation
svm_cv <- train(Results ~ ., data = df, method = "svmLinear", trControl = trainControl(method = "cv", number = 5))

# View the cross-validated results
print(svm_cv)

# Lower RMSE values indicate better model performance, as it means the model's predictions are closer to the actual values on average.


# Confusion matrix after cross validating
test_data[, 1] <- factor(test_data[, 1], levels = levels(svm_pred))

# Create the confusion matrix
con_mat <- confusionMatrix(svm_pred, test_data[, 1])

# Print the confusion matrix
print(con_mat)